classes/observer.php

```diff
@@ -135,4 +137,334 @@ public static function mod_quiz_event_attempt_submitted(
+        if ($eventdata['target'] == 'submission_status') {
+            // The event is triggered when a submission is deleted and when the submission is passed to draft.
+            $fs = get_file_storage();
+            $submissionfiles = $fs->get_area_files(
+                $eventdata["contextid"],
+                "assignsubmission_file",
+                'submission_files',
+                $eventdata["objectid"]
+            );
+
+            // If the documents have been deleted in the mdl_files table, we also delete them on our side.
+            if (empty($submissionfiles)) {
+                // Check if Copyleaks API is connected.
+                $cl = new \plagiarism_copyleaks_comms();
+                $submissionid = $eventdata["objectid"];
+                $data = (array)[
+                    'courseModuleId' => $cmid,
+                    'moodleUserId' => $userid,
+                ];
+                $cl->delete_submission($data, $submissionid); 
+                // $DB->delete_records('plagiarism_copyleaks_files', ['cm' => $cmid, 'userid' => $userid]);
+            }
+        }
+    }
+
+    /**
+     * Handle file deletion in assign submission
+     * @param  \assignsubmission_file\event\submission_updated $event Event
+     * @return void
+     */
+    public static function assign_submission_file_updated(
+        \assignsubmission_file\event\submission_updated $event
+    ) {
+        global $DB;
+
+        // Check if Copyleaks API is connected.
+        if (!plagiarism_copyleaks_dbutils::is_copyleaks_api_connected()) {
+            return;
+        }
+
+        $eventdata = $event->get_data();
+        $cmid = $eventdata["contextinstanceid"];
+        $submissionid = $eventdata["other"]["submissionid"];
+
+        if (!plagiarism_copyleaks_moduleconfig::is_module_enabled('assign', $cmid)) {
+            return;
+        }
+
+        // Get user id.
+        $userid = $eventdata['relateduserid'];
+        if ($userid == null) {
+            $userid = $eventdata['userid'];
+        }
+
+        if ($eventdata['target'] == 'submission' && $eventdata['action'] == 'updated') {
+
+            $clfiles = $DB->get_records('plagiarism_copyleaks_files', ['cm' => $cmid, 'itemid' => $submissionid, 'submissiontype' => 'file']);
+            $fs = get_file_storage();
+            foreach ($clfiles as $clfile) {
+                $file = $fs->get_file_by_hash($clfile->identifier);
+                if ($file === false) {
+                    $cl = new \plagiarism_copyleaks_comms();
+                    $data = (array)[
+                        'identifier' => $clfile->identifier,
+                        'courseModuleId' => $clfile->cm,
+                        'moodleUserId' => $clfile->userid,
+                    ];
+                    $cl->delete_report($data);
+                    // $DB->delete_records('plagiarism_copyleaks_files', ['cm' => $cmid, 'userid' => $clfile->userid, 'identifier' => $clfile->identifier]);
```

Comment:
// $DB->delete_records('plagiarism_copyleaks_files', ['cm' => $cmid, 'userid' => $clfile->userid, 'identifier' => $clfile->identifier]);

@ShadeAma do we still need this? or it can be removed?